philander/mcp40d.py

```diff
@@ -0,0 +1,156 @@
+    The CMP40D17/18/19 chips are digital potentiometers that are controlled via an I2C interface. Their difference lies in their terminal configurations.
+    The all come in different resistances of 5kOhm, 10kOhm, 50kOhm and 100kOhm. Read more under https://www.microship.com/en-us/product/MCP40D17
+    """
+    
+    ADDRESSES_ALLOWED = [0x2E, 0x3E]
+    _potentiometer_resolution = None
+    _potentiometer_resistance_max = None
+
+    def __init__(self):
+        # Create instance attributes and initialize parent classes and interfaces
+        SerialBusDevice.__init__(self)
+        Potentiometer.__init__(self)
+
+    #
+    # Module API
+    #
+
+    @classmethod
+    def Params_init(cls, paramDict):
+        """Initializes configuration parameters with defaults.
+        
+        The following settings are supported:
+        
+        =============================    ==========================================================================================================
+        Key name                         Value type, meaning and default
+        =============================    ==========================================================================================================
+        SerialBusDevice.address          ``int`` I2C serial device address; default is :attr:`ADDRESSES_ALLOWED` [0].
+        Potentiometer.resistance.max     ``int``; Maximum resistance in Ohm; :attr:`DEFAULT_RESISTANCE_MAX`.
+        Potentiometer.resolution         ``int``; Number of possible steps to set resistance value to (2^(bits used for resistance)).
+        ===========================================================================================================================================
+        
+        For the ``SerialBusDevice.address`` value, also 0 or 1
+        can be specified alternatively to the absolute addresses to reflect
+        the level of the ``SDO`` pin. In this case, 0 will be mapped to
+        0x18, while 1 maps to 0x19.
+        
+        Also see: :meth:`.SerialBusDevice.Params_init`, :meth:`.Potentiometer.Params_init`. 
+        """
+        defaults = dict()
+        SerialBusDevice.Params_init( paramDict )
+        Potentiometer.Params_init( paramDict )
+        defaults.update( paramDict )
+        paramDict = defaults
+        return None
+
+    def open(self, paramDict):
+        """Initialize an instance and prepare it for use.
+
+        Also see: :meth:`.Module.open`.
+        
+        :param dict(str, object) paramDict: Configuration parameters as\
+        possibly obtained from :meth:`Params_init`.
+        :return: An error code indicating either success or the reason of failure.
+        :rtype: ErrorCode
+        """
+        # Get default parameters
+        MCP40D.Params_init( paramDict )
+        # Open the bus device
+        ret = SerialBusDevice.open(self, paramDict)
+        # Store potentiometer properties
+        self._potentiometer_resolution = paramDict["Potentiometer.resolution"]
```

Comment:
Define local attributes in the constructor __init__. Set them to e.g. none.
Rename _resolution_ into _digital_max_.